<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of Data Generated by the Tapestri Platform • TapestriR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Analysis of Data Generated by the Tapestri Platform">
<meta property="og:description" content="Detect and Compare cell clones from DNA/Protein analytes, analyse copy number variations from variants detected by Tapestri Pipeline.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">TapestriR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/loading_data.html">Loading Tapestri Data</a>
    </li>
    <li>
      <a href="articles/multiomics_analysis.html">Multi-omics Analysis</a>
    </li>
    <li>
      <a href="articles/raji_k562_analysis.html">Multi-Sample export from Insights</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="overview" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1></div>
<p><code>tapestriR</code> is an R package for DNA + Analysis analysis of data generated by the Tapestri platform</p>
<p>Vignettes showing how to load data and performance analysis:</p>
<p>1) <code>multiomics_analysis.Rmd' - load a multiomic DNA + Protein h5 file.  2)</code>dna_analysis.Rmd’ - load a DNA only loom file. 3) `raji_k562_analysis.Rmd’ - load data from Tapestri Insights export.</p>
</div>
<div id="whats-included" class="section level1">
<h1 class="hasAnchor">
<a href="#whats-included" class="anchor"></a>What’s included</h1>
<ul>
<li>Data manipulation</li>
<li>Read</li>
<li>Filter</li>
<li>Transform</li>
<li>Data Visualization</li>
<li>Object functions</li>
<li>Loading data</li>
</ul>
</div>
<div id="objects" class="section level1">
<h1 class="hasAnchor">
<a href="#objects" class="anchor"></a>Objects</h1>
<p>The multiomics object is a structured way to store all of the raw and normalized data as well as the analysis performed on these data. The object ensures that all of the multiomics data is properly aligned for cells and features of the cells.</p>
<div id="multiomics-object" class="section level3">
<h3 class="hasAnchor">
<a href="#multiomics-object" class="anchor"></a>Multiomics Object</h3>
<pre><code>&lt;root&gt;
+-- metadata                          Experiment meta data. Required to have experiment_name
+-- assays
          +-- dna_variants            DNA Assay object
          +-- dna_read_counts         DNA read counts Assay object
          +-- protein_read_counts     Protein Read counts Assay object
          +--
+-- cell_annotations                  Annotation of each cell, Required to have barcode and sample name columns
</code></pre>
<div id="dna_variants-assay-object" class="section level4">
<h4 class="hasAnchor">
<a href="#dna_variants-assay-object" class="anchor"></a><code>dna_variants</code> Assay Object</h4>
<pre><code>&lt;root&gt;
+-- assay_name                        Name of Assay. Same as one of the layers in the assays slot of Multiomics object
+-- data_layers                       1 or more layers of data. each layer must have same dimensions and matching cell barcodes (rows) and features (columns names) 
                +--NGT                Genotype layer. 0=WT, 1=HET, 2=HOM, 3=unknown
                +--AD
                +--GQ
+-- analysis_layers                   Additional layers after data is processed. Must have same number of cells (rows) 
                    +--               Examples: umap projection of each cell, or kmeans cluster labels
+-- feature_annotations               Annotations of features. Must have id column that is same as column names in data layers.
+-- cell_annotations                  Annotations of cells. Required to have barcode and sample name columns </code></pre>
</div>
<div id="protein_read_counts-assay-object" class="section level4">
<h4 class="hasAnchor">
<a href="#protein_read_counts-assay-object" class="anchor"></a><code>protein_read_counts</code> Assay Object</h4>
<pre><code>&lt;root&gt;
+-- assay_name                        Name of Assay. Same as one of the layers in the assays slot of Multiomics object
+-- metadata                          Name of Assay. Same as one of the layers in the assays slot of Multiomics object
+-- data_layers                       1 or more layers of data. each layer must have same dimensions and matching cell barcodes (rows) and features (columns names) 
                +--read_counts        raw read counts per protein
                +--normalized         normalized read counts
                +--
+-- analysis_layers                   Additional layers after data is processed. Must have same number of cells (rows) 
                    +--
+-- feature_annotations               Annotations of features. Must have id column that is same as column names in data layers.
+-- cell_annotations                  Annotations of cells. Required to have barcode and sample name columns </code></pre>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Mission Bio <br><small class="roles"> Author, maintainer, copyright holder </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Mission Bio.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
