<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Loading Tapestri Data • TapestriR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Loading Tapestri Data">
<meta property="og:description" content="TapestriR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TapestriR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/loading_data.html">Loading Tapestri Data</a>
    </li>
    <li>
      <a href="../articles/multiomics_analysis.html">Multi-omics Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Loading Tapestri Data</h1>
            
      
      
      <div class="hidden name"><code>loading_data.Rmd</code></div>

    </div>

    
    
<div id="load-data-from-various-sources" class="section level1 tabset">
<h1 class="hasAnchor">
<a href="#load-data-from-various-sources" class="anchor"></a>Load data from various sources</h1>
<p>Enable loading from various files.</p>
<ul>
<li>h5 is the future format for miltiomics and multisample</li>
<li>loom is supported for DNA mutation only analysis</li>
<li>Insights export is supported for DNA only analysis</li>
</ul>
<div id="load-from-loom" class="section level2">
<h2 class="hasAnchor">
<a href="#load-from-loom" class="anchor"></a>Load from Loom</h2>
<p>Work for DNA only. Only single sample, since loom files only contain 1 sample.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">filename</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"PE11.cells.loom"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"TapestriR"</span>)

<span class="no">variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_loom.html">read_loom</a></span>(<span class="no">filename</span>,<span class="kw">min_mutation_rate</span> <span class="kw">=</span> <span class="fl">0.05</span>)

<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/filter_variants.html">filter_variants</a></span>(<span class="no">variants</span>)</pre></body></html></div>
<pre><code>## Warning in filter_variants(variants): this method will not return identical
## results to Tapestri Insights. Export filtered results from Tapestri Insights or
## use the Tapestri SDK to filter before importing into R.</code></pre>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">vaf</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">AD</span>/<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">DP</span>, <span class="fl">3</span>)
<span class="no">vaf</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">vaf</span>)] <span class="kw">&lt;-</span> <span class="fl">0</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">filtered_variants</span>,<span class="st">'VAF'</span>,<span class="no">vaf</span>)

<span class="no">filtered_variants</span></pre></body></html></div>
<pre><code>## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "dna"
##   ..@ metadata           :List of 1
##   .. ..$ cell_info: tibble [1 × 2] (S3: tbl_df/tbl/data.frame)
##   ..@ feature_annotations: tibble [8 × 11] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [5,043 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 6
##   .. ..$ NGT: tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   .. ..$ AD : tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   .. ..$ DP : tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   .. ..$ GQ : tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   .. ..$ RO : tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   .. ..$ VAF: tibble [5,043 × 8] (S3: tbl_df/tbl/data.frame)
##   ..@ analysis_layers    : list()</code></pre>
</div>
<div id="load-multiomics-h5" class="section level2">
<h2 class="hasAnchor">
<a href="#load-multiomics-h5" class="anchor"></a>Load multiomics H5</h2>
<p>Works for DNA, DNA + Protein data. Also works from multisample. best practice is to create a multiassay, multisample h5 in pipeline, and apply filters before loading into R.</p>
<p>usage:</p>
<ol style="list-style-type: decimal">
<li>Load variant assay and filter</li>
<li>Load Protein assay and normalize</li>
<li>Load DNA read counts</li>
<li>create a multiomics object and merge all 3 assays. This will make sure each assay has the same cells.</li>
</ol>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">filename</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"ABseq021.h5"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"TapestriR"</span>)

<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_tap.html">read_tap</a></span>(<span class="no">filename</span>)

<span class="co"># ideally would just start by loading filtered H5, but for now will load data with some basic filters</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/filter_variants.html">filter_variants</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>)</pre></body></html></div>
<pre><code>## Warning in filter_variants(experiment$assays$dna): this method will not return
## identical results to Tapestri Insights. Export filtered results from Tapestri
## Insights or use the Tapestri SDK to filter before importing into R.</code></pre>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">vaf</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">AD</span>/<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">DP</span>, <span class="fl">3</span>)
<span class="no">vaf</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">vaf</span>)] <span class="kw">&lt;-</span> <span class="fl">0</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">filtered_variants</span>,<span class="st">'VAF'</span>,<span class="no">vaf</span>)

<span class="co">#add the filtered data back to experiment, this will subset the rest of the assays to make sure we have same cells</span>
<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_assay.html">add_assay</a></span>(<span class="no">experiment</span>,<span class="no">filtered_variants</span>, <span class="kw">keep_common_cells</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co"># normalize using clr method</span>
<span class="no">protein_counts_norm</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">read_counts</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="../reference/clr_by_feature.html">clr_by_feature</a></span>() <span class="kw">%&gt;%</span> <span class="fu">as_tibble</span>(<span class="kw">rownames</span> <span class="kw">=</span> <span class="fl">NA</span>)
<span class="co"># add normalized data to protein assay</span>
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>,<span class="st">'normalized'</span>,<span class="no">protein_counts_norm</span>)


<span class="no">normalized_dna_reads</span> <span class="kw">=</span> <span class="fu"><a href="../reference/normalize_dna_reads.html">normalize_dna_reads</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>$<span class="no">data_layers</span>$<span class="no">read_counts</span>)
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>,<span class="st">'normalized'</span>,<span class="no">normalized_dna_reads</span>)


<span class="no">experiment</span></pre></body></html></div>
<pre><code>## Experiment: ABseq021.h5
## Num Cells: 1197
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "cnv"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [138 × 1] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,197 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 2
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "dna"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [49 × 6] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,197 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 6
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "protein"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [10 × 1] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,197 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 2
##   ..@ analysis_layers    : list()</code></pre>
</div>
<div id="create-your-own-multiomics-object" class="section level2">
<h2 class="hasAnchor">
<a href="#create-your-own-multiomics-object" class="anchor"></a>Create your own multiomics object</h2>
<p>In case you want to build a multiomics object from individual components, not using the <code>read_tap</code> function.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">ASSAY_NAME_VARIANT</span> <span class="kw">=</span> <span class="st">'dna'</span>
<span class="no">ASSAY_NAME_PROTEIN</span> <span class="kw">=</span> <span class="st">'protein'</span>

<span class="no">filename</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"ABseq021.h5"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"TapestriR"</span>)

<span class="co"># ideally would just start by loading filtered H5, but for now will send filtering list to function to mimic</span>
<span class="no">variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_assay_h5.html">read_assay_h5</a></span>(<span class="kw">filename</span> <span class="kw">=</span> <span class="no">filename</span>, <span class="kw">assay_name</span> <span class="kw">=</span> <span class="no">ASSAY_NAME_VARIANT</span>, <span class="kw">min_mutation_rate</span> <span class="kw">=</span> <span class="fl">0.005</span>)

<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/filter_variants.html">filter_variants</a></span>(<span class="no">variants</span>)</pre></body></html></div>
<pre><code>## Warning in filter_variants(variants): this method will not return identical
## results to Tapestri Insights. Export filtered results from Tapestri Insights or
## use the Tapestri SDK to filter before importing into R.</code></pre>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">vaf</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">AD</span>/<span class="no">filtered_variants</span>@<span class="kw">data_layers</span>$<span class="no">DP</span>, <span class="fl">3</span>)
<span class="no">vaf</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">vaf</span>)] <span class="kw">&lt;-</span> <span class="fl">0</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">filtered_variants</span>,<span class="st">'VAF'</span>,<span class="no">vaf</span>)

<span class="co">### load protein assay and normalize</span>
<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_assay_h5.html">read_assay_h5</a></span>(<span class="kw">filename</span> <span class="kw">=</span> <span class="no">filename</span>, <span class="kw">assay_name</span> <span class="kw">=</span> <span class="no">ASSAY_NAME_PROTEIN</span>)
<span class="co"># normalize using clr method</span>
<span class="no">protein_counts_norm</span> <span class="kw">=</span> <span class="no">protein</span>@<span class="kw">data_layers</span>$<span class="no">read_counts</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="../reference/clr_by_feature.html">clr_by_feature</a></span>() <span class="kw">%&gt;%</span> <span class="fu">as_tibble</span>(<span class="kw">rownames</span> <span class="kw">=</span> <span class="fl">NA</span>)
<span class="co"># add normalized data to protein assay</span>
<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">protein</span>,<span class="st">'normalized'</span>,<span class="no">protein_counts_norm</span>)

<span class="co">### create multiomics object (moo)</span>
<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/create_moo.html">create_moo</a></span>(<span class="kw">experiment_name</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span>(<span class="no">filename</span>), <span class="kw">cell_annotations</span> <span class="kw">=</span> <span class="no">filtered_variants</span>@<span class="kw">cell_annotations</span>)
<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_assay.html">add_assay</a></span>(<span class="kw">moo</span> <span class="kw">=</span> <span class="no">experiment</span>,<span class="kw">assay</span> <span class="kw">=</span> <span class="no">filtered_variants</span>)

<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_assay.html">add_assay</a></span>(<span class="no">experiment</span>,<span class="no">protein</span>, <span class="kw">keep_common_cells</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="no">experiment</span></pre></body></html></div>
<pre><code>## Experiment: ABseq021.h5
## Num Cells: 1198
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "dna"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [49 × 6] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,198 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 6
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "protein"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [10 × 1] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,198 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 2
##   ..@ analysis_layers    : list()</code></pre>
<div id="x-y-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#x-y-plots" class="anchor"></a>X-Y plots</h3>
<ol style="list-style-type: decimal">
<li>select which proteins to plot on X and Y axis</li>
<li>select set of other feature(s) to color the plot by. If you choose more then one feature, each feature will be plotted in a subplot</li>
</ol>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co">##################</span>
<span class="co"># select the Proteins to plot on X and Y</span>
<span class="co">##################</span>

<span class="co">#protein_x = 'CD34'</span>
<span class="co">#protein_y = 'CD38'</span>

<span class="no">protein_x</span> <span class="kw">=</span> <span class="fl">1</span>
<span class="no">protein_y</span> <span class="kw">=</span> <span class="fl">2</span>

<span class="co">##################</span>
<span class="co"># select 1 or more features to color by</span>
<span class="co"># color_by should be a vector of column header you want to color by</span>
<span class="co">##################</span>

<span class="co">#all proteins</span>
<span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>

<span class="co">#select a few proteins</span>
<span class="co">#color_by =  experiment$assays$protein$data_layers$normalized %&gt;% select('CD110','CD117')</span>

<span class="co">#select a few variant</span>
<span class="co">#color_by =  experiment$assays$dna$data_layers$NGT %&gt;% select(1:10) %&gt;% mutate_all(as_factor) %&gt;% mutate_all(recode_genotypes)</span>


<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_x</span>]],
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_y</span>]],
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)+ <span class="fu">scale_colour_gradient2</span>(<span class="kw">low</span><span class="kw">=</span><span class="st">"yellow"</span>, <span class="kw">mid</span><span class="kw">=</span><span class="st">'grey'</span>, <span class="kw">high</span><span class="kw">=</span><span class="st">"darkred"</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">xlab</span>(<span class="no">protein_x</span>) + <span class="fu">ylab</span>(<span class="no">protein_y</span>) + <span class="fu">ggtitle</span>(<span class="st">'Color by Protein Expression'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="loading_data_files/figure-html/unnamed-chunk-2-1.png" width="1152"></p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co">#select a few variant</span>
<span class="no">color_by</span> <span class="kw">=</span>  <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">data_layers</span>$<span class="no">NGT</span> <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="fl">1</span>:<span class="fl">10</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">as_factor</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">recode_genotypes</span>)</pre></body></html></div>
<pre><code>## Warning: Unknown levels in `f`: 2</code></pre>
<pre><code>## Warning: Unknown levels in `f`: 3</code></pre>
<pre><code>## Warning: Unknown levels in `f`: 2</code></pre>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_x</span>]],
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_y</span>]],
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">xlab</span>(<span class="no">protein_x</span>) + <span class="fu">ylab</span>(<span class="no">protein_y</span>) + <span class="fu">ggtitle</span>(<span class="st">'Color by Genotypes'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="loading_data_files/figure-html/unnamed-chunk-2-2.png" width="1152"></p>
</div>
</div>
<div id="load-data-from-tapestri-insights-export" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data-from-tapestri-insights-export" class="anchor"></a>Load data from Tapestri Insights Export</h2>
<p>Work for DNA only. Export can be from a multisample project in Tapestri Insights. usage:</p>
<ol style="list-style-type: decimal">
<li>extract exported zip file</li>
<li>read the <code>NGT.csv</code> and <code>Variants.csv</code> files</li>
<li>create new <code>Tapestri_Assay</code> object</li>
</ol>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">export_dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"insights_2.2_export"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"TapestriR"</span>)

<span class="co">#show files</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="no">export_dir</span>)</pre></body></html></div>
<pre><code>## [1] "AF.csv"       "DP.csv"       "GQ.csv"       "NGT.csv"      "README.txt"  
## [6] "Variants.csv"</code></pre>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_insights_export.html">read_insights_export</a></span>(<span class="kw">export_dir</span> <span class="kw">=</span> <span class="no">export_dir</span>)</pre></body></html></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_double(),
##   Sample = col_character()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_character(),
##   DANN = col_double(),
##   `Allele Freq (gnomAD)` = col_double(),
##   Whitelist = col_logical()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">variants</span></pre></body></html></div>
<pre><code>## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "variants"
##   ..@ metadata           :List of 1
##   .. ..$ cell_info: tibble [4 × 2] (S3: tbl_df/tbl/data.frame)
##   ..@ feature_annotations: tibble [23 × 24] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [30,070 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 4
##   .. ..$ NGT: tibble [30,070 × 23] (S3: tbl_df/tbl/data.frame)
##   .. ..$ AF : tibble [30,070 × 23] (S3: tbl_df/tbl/data.frame)
##   .. ..$ DP : tibble [30,070 × 23] (S3: tbl_df/tbl/data.frame)
##   .. ..$ GQ : tibble [30,070 × 23] (S3: tbl_df/tbl/data.frame)
##   ..@ analysis_layers    : list()</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mission Bio.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
