<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Multi-Omics Analysis • TapestriR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Multi-Omics Analysis">
<meta property="og:description" content="TapestriR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TapestriR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/loading_data.html">Loading Tapestri Data</a>
    </li>
    <li>
      <a href="../articles/multiomics_analysis.html">Multi-Omics Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="multiomics_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Multi-Omics Analysis</h1>
            
      
      
      <div class="hidden name"><code>multiomics_analysis.Rmd</code></div>

    </div>

    
    
<div id="load-multiomics-h5" class="section level1">
<h1 class="hasAnchor">
<a href="#load-multiomics-h5" class="anchor"></a>Load multiomics H5</h1>
<p>We will analyze single cell DNA and protein sequencing data from the Tapestri Platform to demonstrate the ability to detect both CNVs and SNVs simultaneously and see how the expression of proteins is changing in different cell populations that are mutant or not.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">#replace filename with path to your data</span>
<span class="no">filename</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"4_cell_line_mix_dna_protein.h5"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"TapestriR"</span>)

<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/read_tap.html">read_tap</a></span>(<span class="no">filename</span>)

<span class="no">experiment</span></pre></body></html></div>
<pre><code>## Experiment: 4_cell_line_mix_dna_protein.h5
## Num Cells: 1239
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "cnv"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [127 × 4] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,239 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 1
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "dna"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [1,089 × 9] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,239 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 5
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "protein"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [10 × 1] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,239 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 1
##   ..@ analysis_layers    : list()</code></pre>
</div>
<div id="filter-variants" class="section level1">
<h1 class="hasAnchor">
<a href="#filter-variants" class="anchor"></a>Filter variants</h1>
<p>Best practice is to create a multi-assay, multi-sample h5 in Pipeline, and apply filters before loading into R. We’ve done some filtering here. Notice how the number of cells and features change after filtering.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co"># Ideally, we would start by loading a filtered H5, but for now, we load data with some basic filters.</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/filter_variants.html">filter_variants</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>)

<span class="co"># Add VAF layer into the DNA assay.</span>
<span class="no">vaf</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="no">filtered_variants</span>$<span class="no">data_layers</span>$<span class="no">AD</span>/<span class="no">filtered_variants</span>$<span class="no">data_layers</span>$<span class="no">DP</span>, <span class="fl">3</span>)
<span class="no">vaf</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">vaf</span>)] <span class="kw">&lt;-</span> <span class="fl">0</span>
<span class="no">filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">filtered_variants</span>,<span class="st">'VAF'</span>,<span class="no">vaf</span>)

<span class="co"># Add the filtered data back to the experiment. This will be a subset of the rest of the assays to ensure that we have the same cells.</span>
<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_assay.html">add_assay</a></span>(<span class="no">experiment</span>,<span class="no">filtered_variants</span>, <span class="kw">keep_common_cells</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="no">experiment</span></pre></body></html></div>
<pre><code>## Experiment: 4_cell_line_mix_dna_protein.h5
## Num Cells: 1233
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "cnv"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [127 × 4] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,233 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 1
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "dna"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [43 × 9] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,233 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 6
##   ..@ analysis_layers    : list()
## Formal class 'Tapestri_Assay' [package "TapestriR"] with 6 slots
##   ..@ assay_name         : chr "protein"
##   ..@ metadata           :List of 1
##   ..@ feature_annotations: tibble [10 × 1] (S3: tbl_df/tbl/data.frame)
##   ..@ cell_annotations   : tibble [1,233 × 3] (S3: tbl_df/tbl/data.frame)
##   ..@ data_layers        :List of 1
##   ..@ analysis_layers    : list()</code></pre>
<div id="filtered-variants" class="section level3">
<h3 class="hasAnchor">
<a href="#filtered-variants" class="anchor"></a>Filtered variants</h3>
<p>Picking the right set of variants is critical.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">#manually select the rest of variants</span>
<span class="no">good_variants</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">feature_annotations</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">QUAL</span> <span class="kw">&gt;</span> <span class="fl">30000</span>)

<span class="co">#Manual filters</span>
<span class="no">manually_filtered_variants</span> <span class="kw">=</span> <span class="fu"><a href="../reference/subset_assay.html">subset_assay</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>, <span class="kw">keep_feature_ids</span> <span class="kw">=</span> <span class="no">good_variants</span>$<span class="no">id</span>)
<span class="no">experiment</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_assay.html">add_assay</a></span>(<span class="no">experiment</span>,<span class="no">manually_filtered_variants</span>, <span class="kw">keep_common_cells</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="co">#DT::datatable(experiment$assays$dna$feature_annotations)</span></pre></body></html></div>
</div>
</div>
<div id="normalize-the-read-counts" class="section level1">
<h1 class="hasAnchor">
<a href="#normalize-the-read-counts" class="anchor"></a>Normalize the read counts</h1>
<ol style="list-style-type: decimal">
<li>Load Protein assay and normalize.</li>
<li>Load DNA read counts.</li>
</ol>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co"># Normalize the protein data using the clr method.</span>
<span class="no">protein_counts_norm</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">read_counts</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="../reference/clr_by_feature.html">clr_by_feature</a></span>() <span class="kw">%&gt;%</span> <span class="fu">as_tibble</span>(<span class="kw">rownames</span> <span class="kw">=</span> <span class="fl">NA</span>)

<span class="co"># Add the normalized data to the protein assay.</span>
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>,<span class="st">'normalized'</span>,<span class="no">protein_counts_norm</span>)

<span class="co"># Normalize the DNA read counts. </span>
<span class="no">normalized_dna_reads</span> <span class="kw">=</span> <span class="fu"><a href="../reference/normalize_dna_reads.html">normalize_dna_reads</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>$<span class="no">data_layers</span>$<span class="no">read_counts</span>)
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_data_layer.html">add_data_layer</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>,<span class="st">'normalized'</span>,<span class="no">normalized_dna_reads</span>)</pre></body></html></div>
</div>
<div id="explore-the-data" class="section level1 tabset">
<h1 class="hasAnchor">
<a href="#explore-the-data" class="anchor"></a>Explore the data</h1>
<div id="x-y-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#x-y-plots" class="anchor"></a>X-Y plots</h2>
<ol style="list-style-type: decimal">
<li>Select the proteins to plot on the X and Y axes.</li>
<li>Select the set of other feature(s) to color the plot by. If you choose more than one feature, each feature will be plotted in a subplot.</li>
</ol>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">##################</span>
<span class="co"># Select the proteins to plot on X and Y.</span>
<span class="co">##################</span>

<span class="co"># protein_x = 'CD34'</span>
<span class="co"># protein_y = 'CD38'</span>

<span class="no">protein_x</span> <span class="kw">=</span> <span class="fl">1</span>
<span class="no">protein_y</span> <span class="kw">=</span> <span class="fl">2</span>

<span class="co">##################</span>
<span class="co"># Select 1 or more features to color by.</span>
<span class="co"># color_by should be a vector of the column header you want to color by.</span>
<span class="co">##################</span>

<span class="co"># All proteins</span>
<span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>

<span class="co"># Select a few proteins.</span>
<span class="co"># color_by =  experiment$assays$protein$data_layers$normalized %&gt;% select('CD110','CD117')</span>

<span class="co"># Select a few variants.</span>
<span class="co"># color_by =  experiment$assays$dna$data_layers$NGT %&gt;% select(1:10) %&gt;% mutate_all(as_factor) %&gt;% mutate_all(recode_genotypes)</span>


<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_x</span>]],
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_y</span>]],
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)+ <span class="fu">scale_colour_gradient2</span>(<span class="kw">low</span><span class="kw">=</span><span class="st">"yellow"</span>, <span class="kw">mid</span><span class="kw">=</span><span class="st">'grey'</span>, <span class="kw">high</span><span class="kw">=</span><span class="st">"darkred"</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">xlab</span>(<span class="no">protein_x</span>) + <span class="fu">ylab</span>(<span class="no">protein_y</span>) + <span class="fu">ggtitle</span>(<span class="st">'Color by Protein Expression'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-2-1.png" width="1152"></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co"># Select a few variants.</span>
<span class="no">color_by</span> <span class="kw">=</span>  <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">data_layers</span>$<span class="no">NGT</span> <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="fl">1</span>:<span class="fl">10</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">as_factor</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">recode_genotypes</span>)


<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_x</span>]],
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span><span class="kw">[[</span><span class="no">protein_y</span>]],
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">xlab</span>(<span class="no">protein_x</span>) + <span class="fu">ylab</span>(<span class="no">protein_y</span>) + <span class="fu">ggtitle</span>(<span class="st">'Color by Genotypes'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-2-2.png" width="1152"></p>
</div>
<div id="snv" class="section level2">
<h2 class="hasAnchor">
<a href="#snv" class="anchor"></a>SNV</h2>
<p>We will start with the genotype data and explore also how we can identify clones</p>
<p>###determine clones similar to Tapestri Insights</p>
<p>based on genotypes, group each cell into subclones as Tapestri Insights does.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">#define subclones based on NGT values similar to TI</span>
<span class="no">ngt_clones</span> <span class="kw">=</span> <span class="fu"><a href="../reference/define_subclones.html">define_subclones</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>, <span class="kw">ignore_zygosity</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>, <span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'TI subclone'</span>, <span class="no">ngt_clones</span>$<span class="no">subclone_label</span>)</pre></body></html></div>
<div id="umap-projection" class="section level3">
<h3 class="hasAnchor">
<a href="#umap-projection" class="anchor"></a>Umap projection</h3>
<p>First, let’s reduce the dimensions and see how this will be displayed in a UMAP plot.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">data</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">data_layers</span>$<span class="no">VAF</span>

<span class="no">projections</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="co"># Dimensional reduction using umap.</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">111</span>)
<span class="no">umap_values</span> <span class="kw">&lt;-</span> <span class="fu">umap</span>(<span class="no">data</span>, <span class="kw">scale</span><span class="kw">=</span><span class="fl">TRUE</span>, <span class="kw">metric</span><span class="kw">=</span><span class="st">"manhattan"</span>, <span class="kw">init</span><span class="kw">=</span><span class="st">"laplacian"</span>, <span class="kw">pca</span><span class="kw">=</span><span class="fl">20</span>)
<span class="no">projections</span><span class="kw">[[</span><span class="st">'Projection:umap_manhattan DR:pca data:vaf'</span>]] <span class="kw">=</span> <span class="fu">tibble</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">umap_values</span>[,<span class="fl">1</span>], <span class="kw">y</span> <span class="kw">=</span> <span class="no">umap_values</span>[,<span class="fl">2</span>])

<span class="co"># umap_values &lt;- umap(data, scale=TRUE, metric="euclidean", init="laplacian", pca=20) </span>
<span class="co"># projections[['Projection:umap_euclidean DR:pca data:vaf']] = tibble(x = umap_values[,1], y = umap_values[,2])</span>
<span class="co"># </span>
<span class="co"># umap_values &lt;- umap(data, scale=TRUE, metric="cosine", init="laplacian", pca=20) </span>
<span class="co"># projections[['Projection:umap_cosine DR:pca data:vaf']] = tibble(x = umap_values[,1], y = umap_values[,2])</span>

<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>, <span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'projections'</span>, <span class="fu">as_tibble</span>(<span class="no">projections</span>))</pre></body></html></div>
<p>Visualize the projection</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co"># Show a simple plot of the projection.</span>
<span class="no">projection</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>
<span class="fu">ggplot</span>(<span class="kw">data</span><span class="kw">=</span><span class="no">projection</span>) + <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">x</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">y</span>), <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">size</span><span class="kw">=</span><span class="fl">0.8</span>)</pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/umap-1.png" width="700"></p>
</div>
<div id="determine-the-number-of-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#determine-the-number-of-clusters" class="anchor"></a>Determine the number of clusters</h3>
<p>In this step, we determine the optimal number of clusters for our dataset using k-means or hierarchical clustering. This is not a trivial problem. It is left to the user to explore the data. Here are a few examples for determining the number of clusters in your data.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">cluster_on</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>

<span class="co"># Elbow method</span>
<span class="no">elbow</span> <span class="kw">=</span> <span class="fu">fviz_nbclust</span>(<span class="no">cluster_on</span>, <span class="no">kmeans</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"wss"</span>) +
  <span class="fu">labs</span>(<span class="kw">subtitle</span> <span class="kw">=</span> <span class="st">"Elbow method"</span>)

<span class="co"># Silhouette method</span>
<span class="no">silhouette</span> <span class="kw">=</span> <span class="fu">fviz_nbclust</span>(<span class="no">cluster_on</span>, <span class="no">kmeans</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"silhouette"</span>)+
  <span class="fu">labs</span>(<span class="kw">subtitle</span> <span class="kw">=</span> <span class="st">"Silhouette method"</span>)

<span class="co"># Gap statistic</span>
<span class="co"># nboot = 50 to keep the function speedy. </span>
<span class="co"># Recommended value: nboot= 500 for your analysis</span>
<span class="co"># Use verbose = FALSE to hide computing progression</span>
<span class="co"># set.seed(123)</span>
<span class="co"># gap_stat = fviz_nbclust(cluster_on, kmeans, nstart = 25,  method = "gap_stat", nboot = 25)+</span>
<span class="co">#   labs(subtitle = "Gap statistic method")</span>


(<span class="no">elbow</span> + <span class="no">silhouette</span>) <span class="co">#/ </span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/num_clusters-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><html><body><pre class="r">  # (gap_stat + plot_spacer())</pre></body></html></div>
</div>
<div id="cluster-on-umap-projection" class="section level3">
<h3 class="hasAnchor">
<a href="#cluster-on-umap-projection" class="anchor"></a>Cluster on umap projection</h3>
<p>Clustering the data is a not a trival task. * Clustering can be done one umap projection or the raw data. * There are numerous clustering methods with parameters that need optimization for your data</p>
<p>It is left to use to explore the rich community of R packages for clustering. Here we show clustering by kmeans on a umap projection.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co"># Hold all the different customer labels in a single structure.</span>

<span class="no">cluster_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>
<span class="no">clusters</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="co"># Do the clustering.</span>

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="fl">2</span>:<span class="fl">10</span>) {
  <span class="no">kmean_values</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span>(<span class="no">cluster_by</span>, <span class="no">i</span> ,<span class="kw">iter.max</span><span class="kw">=</span><span class="fl">500</span>)
  <span class="no">clusters</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'umap.kmean.cluster.'</span>,<span class="no">i</span>)]] <span class="kw">=</span> <span class="fu">as_factor</span>(<span class="no">kmean_values</span>$<span class="no">cluster</span>)
}

<span class="co">#############</span>
<span class="co"># Add cluster labels to the analysis data structure.</span>
<span class="co">#############</span>
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>, <span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'umap_vaf_clusters'</span>, <span class="fu">as_tibble</span>(<span class="no">clusters</span>))</pre></body></html></div>
</div>
<div id="compare-different-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-different-clusters" class="anchor"></a>Compare different clusters</h3>
<p>In this example we’re plotting the TI subclone labels on the the umap projection on VAF data.</p>
<p>color_by can be a data.frame of multiple clustering methods and all will be plotted side by side</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">`TI subclone`</span>
<span class="no">projection</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>


<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">projection</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">projection</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>() + <span class="fu">ggtitle</span>(<span class="st">'umap_vaf_clusters'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-4-1.png" width="1152"></p>
</div>
<div id="color-the-umap-by-genotypes" class="section level3">
<h3 class="hasAnchor">
<a href="#color-the-umap-by-genotypes" class="anchor"></a>Color the UMAP by genotypes</h3>
<div class="sourceCode" id="cb16"><html><body><pre class="r"> <span class="co"># %&gt;% select(!contains('chr2:198267'))</span>

<span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">data_layers</span>$<span class="no">NGT</span> <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="fl">1</span>:<span class="fl">20</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">as_factor</span>) <span class="kw">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="no">recode_genotypes</span>)
<span class="no">projection</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>

<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">projection</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">projection</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>()
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">ggtitle</span>(<span class="st">'Color by Genotypes'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-5-1.png" width="1152"></p>
</div>
<div id="violin-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#violin-plots" class="anchor"></a>Violin plots</h3>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>
<span class="no">projection</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>

<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">projection</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">projection</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>() + <span class="fu">ggtitle</span>(<span class="st">'umap_vaf_clusters'</span>) + <span class="fu">theme</span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">'none'</span>)


<span class="no">v</span> <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_violinplot.html">tapestri_violinplot</a></span>(<span class="kw">clusters</span> <span class="kw">=</span> <span class="no">color_by</span>,
               <span class="kw">features</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>)
<span class="no">v</span> <span class="kw">=</span> <span class="no">v</span> + <span class="fu">theme_bw</span>() + <span class="fu">theme</span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>,
                            <span class="kw">axis.text.x</span> <span class="kw">=</span> <span class="fu">element_text</span>(<span class="kw">angle</span> <span class="kw">=</span> <span class="fl">90</span>, <span class="kw">hjust</span> <span class="kw">=</span> <span class="fl">1</span>))

<span class="co"># pathwork magic</span>
<span class="no">p</span> / <span class="no">v</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-6-1.png" width="1152"></p>
</div>
<div id="color-the-umap-by-proteins" class="section level3">
<h3 class="hasAnchor">
<a href="#color-the-umap-by-proteins" class="anchor"></a>Color the UMAP by proteins</h3>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="no">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>
<span class="no">projection</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">projections</span>$<span class="no">`Projection:umap_manhattan DR:pca data:vaf`</span>


<span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">projection</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">projection</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">color_by</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>() + <span class="fu">scale_colour_gradient2</span>(<span class="kw">low</span><span class="kw">=</span><span class="st">"yellow"</span>, <span class="kw">mid</span><span class="kw">=</span><span class="st">'grey'</span>, <span class="kw">high</span><span class="kw">=</span><span class="st">"darkred"</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">ggtitle</span>(<span class="st">'Color by Proteins'</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-7-1.png" width="1152"></p>
</div>
<div id="single-assay-heatmaps" class="section level3">
<h3 class="hasAnchor">
<a href="#single-assay-heatmaps" class="anchor"></a>Single assay heatmaps</h3>
<p>Next we plot the SNVs for each filtered variant across all cells in heatmap format. We’re only providing a simple example to get started.</p>
<p>Users should become familiar with ComplexHeatmap::Heatmap <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/" class="uri">https://jokergoo.github.io/ComplexHeatmap-reference/book/</a>.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="co"># Order the features in chr order.</span>
<span class="no">variant_order</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">feature_annotations</span> <span class="kw">%&gt;%</span>
  <span class="fu">mutate</span>(<span class="kw">CHROM</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="no">CHROM</span>), <span class="kw">POS</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="no">POS</span>)) <span class="kw">%&gt;%</span>
  <span class="fu">arrange</span>(<span class="no">CHROM</span>, <span class="no">POS</span>)

<span class="no">genotypes.mat</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>@<span class="kw">data_layers</span>$<span class="no">NGT</span> <span class="kw">%&gt;%</span> <span class="fu">select</span>(<span class="no">variant_order</span>$<span class="no">id</span>)

<span class="no">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>@<span class="kw">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>

<span class="no">snv.h</span> <span class="kw">&lt;-</span> <span class="kw pkg">ComplexHeatmap</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">genotypes.mat</span>),
  <span class="kw">name</span> <span class="kw">=</span> <span class="st">"GT"</span>,
  <span class="kw">col</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"lightgrey"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span>, <span class="st">"black"</span>),
  <span class="co">#circlize::colorRamp2(c(0, 1, 2, 3), c("grey", "yellow", "blue", "black"))</span>

  <span class="kw">heatmap_legend_param</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">labels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"WT"</span>, <span class="st">"HET"</span>, <span class="st">"HOM"</span>, <span class="st">"Missing"</span>)),

  <span class="kw">split</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">clusters</span>),

  <span class="kw">cluster_rows</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">show_row_names</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">cluster_columns</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">row_title_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span> <span class="kw">=</span> <span class="fl">6</span>),
  <span class="kw">column_names_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span> <span class="kw">=</span> <span class="fl">8</span>),
  <span class="kw">show_column_dend</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)

<span class="no">snv.h</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
<div id="protein" class="section level2">
<h2 class="hasAnchor">
<a href="#protein" class="anchor"></a>Protein</h2>
<div id="cluster-by-proteins" class="section level3">
<h3 class="hasAnchor">
<a href="#cluster-by-proteins" class="anchor"></a>Cluster by proteins</h3>
<p>Let’s explore our protein count data. As we identify 4 clusters using the genotype data, we define also here 4 clones too.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="co"># Dimensional reduction using umap</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">111</span>)
<span class="no">umap_values</span> <span class="kw">&lt;-</span> <span class="fu">umap</span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>, <span class="kw">scale</span><span class="kw">=</span><span class="fl">TRUE</span>, <span class="kw">metric</span><span class="kw">=</span><span class="st">"euclidean"</span>, <span class="kw">init</span><span class="kw">=</span><span class="st">"laplacian"</span>, <span class="kw">pca</span><span class="kw">=</span><span class="fl">20</span>)

<span class="no">umap_layer</span> <span class="kw">=</span> <span class="fu">tibble</span>(
      <span class="kw">x</span> <span class="kw">=</span> <span class="no">umap_values</span>[,<span class="fl">1</span>],
      <span class="kw">y</span> <span class="kw">=</span> <span class="no">umap_values</span>[,<span class="fl">2</span>]
)

<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>, <span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'umap'</span>, <span class="no">umap_layer</span>)

<span class="co"># Hold all the different customer labels in a single structure.</span>
<span class="co"># cluster_by = experiment$assays$protein$analysis_layers$umap</span>
<span class="no">cluster_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>

<span class="no">clusters</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()

<span class="co"># Do the clustering.</span>

<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="fl">2</span>:<span class="fl">10</span>) {
  <span class="no">kmean_values</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span>(<span class="no">cluster_by</span>, <span class="no">i</span> ,<span class="kw">iter.max</span><span class="kw">=</span><span class="fl">500</span>)
  <span class="no">clusters</span><span class="kw">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'umap.kmean.cluster.'</span>,<span class="no">i</span>)]] <span class="kw">=</span> <span class="fu">as_factor</span>(<span class="no">kmean_values</span>$<span class="no">cluster</span>)

}

<span class="co"># Add cluster labels to the analysis data structure.</span>
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span> <span class="kw">=</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>, <span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'clusters'</span>, <span class="fu">as_tibble</span>(<span class="no">clusters</span>))</pre></body></html></div>
</div>
<div id="plot-the-umap-and-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-the-umap-and-clusters" class="anchor"></a>Plot the UMAP and clusters</h3>
<p>Plot all the different clusters on the same umap projection.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">clusters</span>)

<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu">xlab</span>(<span class="st">''</span>) + <span class="fu">ylab</span>(<span class="st">''</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>()
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-9-1.png" width="1152"></p>
</div>
<div id="violin-plots-1" class="section level3">
<h3 class="hasAnchor">
<a href="#violin-plots-1" class="anchor"></a>Violin plots</h3>
<p>To do:</p>
<ul>
<li>Name the clusters based on the signature.</li>
<li>Show the heatmap of the signature.</li>
</ul>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">p_umap</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">clusters</span>$<span class="no">umap.kmean.cluster.4</span>) +
  <span class="fu">xlab</span>(<span class="st">''</span>) + <span class="fu">ylab</span>(<span class="st">''</span>) + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>()

<span class="no">p_violin</span> <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_violinplot.html">tapestri_violinplot</a></span>(
           <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">clusters</span>$<span class="no">umap.kmean.cluster.4</span>,
           <span class="kw">features</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>)

<span class="no">p_umap</span> / <span class="no">p_violin</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-10-1.png" width="1152"></p>
</div>
<div id="color-umap-by-features" class="section level3">
<h3 class="hasAnchor">
<a href="#color-umap-by-features" class="anchor"></a>Color UMAP by features</h3>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">p</span>  <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_scatterplot.html">tapestri_scatterplot</a></span>(
                 <span class="kw">x</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">x</span>,
                 <span class="kw">y</span><span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">analysis_layers</span>$<span class="no">umap</span>$<span class="no">y</span>,
                 <span class="kw">color_by</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>)
<span class="no">p</span> <span class="kw">=</span> <span class="no">p</span> + <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span>() + <span class="fu">scale_colour_gradient2</span>(<span class="kw">low</span><span class="kw">=</span><span class="st">"yellow"</span>, <span class="kw">mid</span><span class="kw">=</span><span class="st">'grey'</span>, <span class="kw">high</span><span class="kw">=</span><span class="st">"darkred"</span>)
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-11-1.png" width="1152"> ### Single assay heatmaps</p>
<p>To do:</p>
<ul>
<li>Provide a way to sort within each cluster.</li>
<li>Decide what to do with multi-sample.</li>
<li>Remove features that are not informative.</li>
<li>Label analytes.</li>
</ul>
<p>Users should become familiar with ComplexHeatmap::Heatmap <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/" class="uri">https://jokergoo.github.io/ComplexHeatmap-reference/book/</a>. We’re only providing a simple example to get started.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="co"># Order the features in chr order.</span>
<span class="no">protein.mat</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">protein</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>
<span class="no">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>@<span class="kw">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>



<span class="no">protein.h</span> <span class="kw">&lt;-</span> <span class="kw pkg">ComplexHeatmap</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">protein.mat</span>),
  <span class="kw">name</span> <span class="kw">=</span> <span class="st">"Protein"</span>,
  <span class="kw">cluster_rows</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">cluster_columns</span> <span class="kw">=</span> <span class="fl">FALSE</span>,

  <span class="kw">col</span> <span class="kw">=</span> <span class="kw pkg">circlize</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html">colorRamp2</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(-<span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">2</span>), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"yellow"</span>, <span class="st">"grey"</span>, <span class="st">"blue"</span>)),

  <span class="kw">split</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">clusters</span>),
  <span class="kw">show_row_names</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">row_title_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span> <span class="kw">=</span> <span class="fl">6</span>),
  <span class="co">#heatmap_legend_param=legend_params,</span>
  <span class="kw">column_names_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span> <span class="kw">=</span> <span class="fl">8</span>),
  <span class="kw">show_column_dend</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)

<span class="no">protein.h</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
</div>
<div id="cnv" class="section level2">
<h2 class="hasAnchor">
<a href="#cnv" class="anchor"></a>CNV</h2>
<div id="calculate-ploidy" class="section level3">
<h3 class="hasAnchor">
<a href="#calculate-ploidy" class="anchor"></a>Calculate ploidy</h3>
<p>Usage:</p>
<ul>
<li>Select a cluster that you think is the normal population. ** Explore your data to identify a control group, which is assumed to be normal population and therefore deploid.</li>
<li>Normalize the read counts based on this normal population.</li>
<li>Plot as a heatmap to review.</li>
</ul>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="co"># Select a clustering method that best represents your data and identify the normal cluster.</span>
<span class="co"># Normalize all the read counts based on the normal cluster.</span>
<span class="no">cnv.mat</span> <span class="kw">=</span> <span class="fu"><a href="../reference/compute_ploidy.html">compute_ploidy</a></span>(
  <span class="kw">reads</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>$<span class="no">data_layers</span>$<span class="no">normalized</span>,
  <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>,
  <span class="kw">baseline_cluster</span> <span class="kw">=</span> <span class="fl">3</span>
 )
<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/add_analysis_layer.html">add_analysis_layer</a></span>(<span class="kw">assay</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>,<span class="kw">layer_name</span> <span class="kw">=</span> <span class="st">'norm_to_baseline'</span>,<span class="kw">data</span> <span class="kw">=</span> <span class="no">cnv.mat</span>)</pre></body></html></div>
</div>
<div id="heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>Heatmap</h3>
<p>From the heatmap we can clearly see the regions where there is loss of heterozygosity and ploidy from 2 is reduced closer to 1.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">cnv.mat</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>$<span class="no">analysis_layers</span>$<span class="no">norm_to_baseline</span>)
<span class="no">cnv.mat</span>[<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">cnv.mat</span>) <span class="kw">|</span> <span class="fu"><a href="https://rdrr.io/r/base/is.finite.html">is.infinite</a></span>(<span class="no">cnv.mat</span>)] <span class="kw">&lt;-</span><span class="fl">2</span>

<span class="no">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>@<span class="kw">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>

<span class="no">cnv.h</span> <span class="kw">&lt;-</span> <span class="kw pkg">ComplexHeatmap</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(
   <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="no">cnv.mat</span>),
   <span class="kw">name</span> <span class="kw">=</span> <span class="st">"CNV"</span>,
   <span class="kw">col</span> <span class="kw">=</span> <span class="kw pkg">circlize</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html">colorRamp2</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">4</span>), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"yellow"</span>, <span class="st">"grey"</span>, <span class="st">"blue"</span>)),
   <span class="kw">split</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">clusters</span>),

   <span class="kw">cluster_rows</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
   <span class="kw">show_row_names</span><span class="kw">=</span><span class="fl">FALSE</span>,
   <span class="kw">cluster_columns</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
   <span class="kw">row_title_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span> <span class="kw">=</span> <span class="fl">6</span>),
   <span class="kw">column_names_gp</span> <span class="kw">=</span> <span class="kw pkg">grid</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html">gpar</a></span>(<span class="kw">fontsize</span><span class="kw">=</span><span class="fl">8</span>),
   <span class="kw">show_column_dend</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">cnv.h</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div id="ploidy-line-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#ploidy-line-plot" class="anchor"></a>Ploidy line plot</h3>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="no">p</span> <span class="kw">=</span> <span class="fu"><a href="../reference/tapestri_ploidy_plot.html">tapestri_ploidy_plot</a></span>(
  <span class="kw">normalized_reads</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">cnv</span>$<span class="no">analysis_layers</span>$<span class="no">norm_to_baseline</span>,
  <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">experiment</span>$<span class="no">assays</span>$<span class="no">dna</span>$<span class="no">analysis_layers</span>$<span class="no">umap_vaf_clusters</span>$<span class="no">umap.kmean.cluster.4</span>
  )
<span class="no">p</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
</div>
<div id="multi-omic-heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#multi-omic-heatmap" class="anchor"></a>Multi-omic heatmap</h2>
<p>Combine SNV, CNV and Protein data into a single heatmap. A nice feature of Complexheatmaps package</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="no">snv.h</span> + <span class="no">cnv.h</span> + <span class="no">protein.h</span></pre></body></html></div>
<p><img src="multiomics_analysis_files/figure-html/unnamed-chunk-16-1.png" width="1152"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mission Bio.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
